crates/consensus/src/receipt/receipts.rs

```diff
@@ -42,7 +42,10 @@ where
     }
 }
 
-impl TxReceipt for Receipt {
+impl<T> TxReceipt<T> for Receipt<T>
+where
+    T: Borrow<Log>,
```

Comment:
do we want this? thought we may want to be generic over this and do `T = Log` perhaps

Reply:
```rust 

pub fn bloom_slow(&self) -> Bloom {
        self.logs.iter().map(Borrow::borrow).collect()
}

```

We made use of this here; 
We might have to change this implementation, thou `Log` works for most of the use-cases I can think of now.

Reply:
it's a bit more complicated and we will need more work here. Because `Log` is actually `Log<T = LogData>` we want to be able to implement generically across a wide variety of types, and may need to introduce a `trait Log` in order to do it

For now, this is an improvement over the existing state of things so we might as well